lib/buildenv/default.nix

```diff
@@ -0,0 +1,84 @@
+# buildEnv creates a tree of symlinks to the specified paths.  This is
+# a fork of the buildEnv in the Nix distribution.  Most changes should
+# eventually be merged back into the Nix distribution.
```

Comment:
I would note what the differences are and make it clear that this should not be used anywhere other than a flox environment for the time being

Reply:
No .. this is an *exact* copy of buildenv from nixpkgs upstream, and we're not changing anything in these files to make it easier to backport updates from upstream over time.

Reply:
I don't think merging a `# flox fork: ...` comment at line 1 would be a difficult merge conflict to resolve or a difficult line to disregard when diffing - not sure if there's something else you have in mind?

Reply:
Discussed with Matthew offline that we added a README.flox to make clear how these files were created and updated over time.